merkle/src/store.rs

```diff
@@ -0,0 +1,411 @@
+use merkle::Element;
+use memmap::MmapMut;
+use memmap::MmapOptions;
+use positioned_io::{ReadAt, WriteAt};
+use std::fs::File;
+use std::marker::PhantomData;
+use std::ops::{self, Index};
+use tempfile::tempfile;
+
+
+/// Backing store of the merkle tree.
+pub trait Store<E: Element>:
+    ops::Deref<Target = [E]> + std::fmt::Debug + Clone + Send + Sync
+{
+    /// Creates a new store which can store up to `size` elements.
+    fn new(size: usize) -> Result<Self, &'static str>;
```

Comment:
more of a Rust question than a review comment: why use static strings here? is it for performance? asking since I haven't seen this type of error return signature

Reply:
No strong reason, still learning.

Reply:
> Definitely in the right direction but I don't think this fixes issue #19 though, mostly because that issue is outdated (I don't think new_with_path even exists anymore).

Correct, I meant `new_from_slice`, but typo'd it based on the issue name.

Reply:
Ok, in that case you can use patterns like https://github.com/filecoin-project/rust-fil-proofs/blob/master/filecoin-proofs/src/error.rs in the future.